lib/rubocop/cop/rails/to_s_with_argument.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Identifies passing any argument to `#to_s`.
+      #
+      # @example
+      #
+      #   # bad
+      #   to_s(:delimited)
+      #
+      #   # good
+      #   to_formatted_s(:delimited)
+      #
+      class ToSWithArgument < Base
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        MSG = 'Use `to_formatted_s(...)` instead of `to_s(...)`.'
+
+        RESTRICT_ON_SEND = %i[to_s].freeze
+
+        minimum_target_rails_version 7.0
+
+        def on_send(node)
+          return if node.arguments.empty?
+
+          add_offense(node) do |rewriter|
+            rewriter.replace(
+              node.loc.selector,
+              'to_formatted_s'
+            )
+          end
```

Comment:
That's my two cents. Can you tweak it?

```suggestion
          add_offense(node) do |corrector|
            corrector.replace(node.loc.selector, 'to_formatted_s')
          end
```